src/charm.py

```diff
@@ -320,27 +311,48 @@ def _reset_environment() -> None:
         os.mkdir(CONFIG_PATH)
         os.mkdir(SERVICE_PATH)
 
-    def _get_bind_address(self) -> Optional[str]:
-        """Returns bind address."""
-        return (
-            ip_from_iface(bind_address_subnet)
-            if (bind_address_subnet := self.model.config.get("bind-address-subnet"))
-            else ip_from_default_iface()
-        )
-
-    def _get_current_status(self) -> ActiveStatus:
-        """Returns current status."""
-        status_type = ActiveStatus
+    @property
+    def _active_status_msg(self) -> str:
+        """Returns msg of current status."""
         status_msg = ""
-        if self._stored.installed:
+        if self._get_peer_data("installed"):
             status_msg = "SW installed."
-        if self._stored.started and service_active(SRS_ENB_SERVICE):
+        if self._get_peer_data("started") and service_active(SRS_ENB_SERVICE):
             status_msg = "srsenb started. "
-            if mme_addr := self._stored.mme_addr:
+            if mme_addr := self._mme_addr:
                 status_msg += f"mme: {mme_addr}. "
-            if self._stored.ue_attached and service_active(SRS_UE_SERVICE):
+            if self._get_peer_data("ue_attached") and service_active(SRS_UE_SERVICE):
                 status_msg += "ue attached. "
-        return status_type(status_msg)
+        return status_msg
+
+    @property
+    def _mme_addr(self) -> Optional[str]:
+        """Returns the ipv4 address of the mme interface.
+
+        Returns:
+            str: mme_addr
+        """
+        if not self._get_peer_data("mme_addr"):
+            return None
+        return self._get_peer_data("mme_addr")
+
+    @property
+    def _bind_address(self) -> Optional[str]:
+        """Returns bind address."""
+        return (
+            bind_address
+            if (bind_address := self.model.config.get("bind-address"))
+            else ip_from_default_iface()
+        )
+
+    def _set_peer_data(self, key: str, data: Any) -> None:
+        """Put information into the peer data bucket instead of `StoredState`."""
+        self.model.get_relation("replicas").data[self.app][key] = json.dumps(data)  # type: ignore[union-attr]  # noqa: E501
+
+    def _get_peer_data(self, key: str) -> Any:
+        """Retrieve information from the peer data bucket instead of `StoredState`."""
```

Comment:
Do we need to know the charm was previously using StoredState?
```suggestion
        """Retrieve information from the peer data bucket."""
```

Reply:
I removed the function completely